scanner_test.go

```diff
@@ -46,6 +46,22 @@ func TestScannerLongLine(t *testing.T) {
 	}
 }
 
+func TestLargePayload(t *testing.T) {
+
+	ctx := context.Background()
+	payload := `{"msg":` + strings.Repeat("a", 1024*1024) + `}` // more than 1mb long json payload
```

Comment:
```suggestion
	payload := `{"msg":` + strings.Repeat("a", 2*1024*1024) + `}` // more than 1mb long json payload
```
shouldn't this be large enough to overflow the buffer?

also, probably better to use the `bufferMaxSize` constant
```suggestion
	payload := `{"msg":` + strings.Repeat("a", bufferMaxSize + 1) + `}` // more than 1mb long json payload
```